design/read-only-registry.md

```diff
@@ -0,0 +1,174 @@
+This flag will be app specific since registry settings are tied to the application.
+
+#### Online Admin Console
+
+This is the most common use of kots install command.
+It results in kotsadm being installed in online mode, while the application can be either online or airgapped:
+
+```
+kubectl kots install app-slug --disable-image-push
+```
+
+When the above command is used, the user will go through the existing installation flow without any changes from the user's perspective.
+Internally, the functionality will change as follows:
+- If airgap option is selected, images will **not** be pushed to the private registry specified in the UI.
+- If online option is selected, updating Registry Settings page will **not** result in images being pushed.
+
+The Registry Settings page is discussed in more detail below.
+
+#### Airgapped Admin Console
+
+In this case, Admin Console images are pulled from a private registry, while the application can be either online or airgapped:
+
+```
+kubectl kots install app-slug \
+  --kotsadm-registry my.private.registry/myapp \
+  --registry-username <username> \
+  --registry-password <password> \
+  --disable-image-push
+```
+
+When the above command is used, the following will happen:
+
+1. All kotsadm pods will be rewritten to pull images from the private registry. This is existing functionality.
+1. Provided registry credentials will be saved in `kotsadm-private-registry` secret in the app's namespace. This is existing functionality.
+1. The `kots admin-console push-images` command can be ommitted.
+1. During license upload in Admin Console UI, registry settings will be populated with these values.
+1. After the application is installed, Registry Settings page will be populated with these values.
+
+During application installation:
+- If airgap option is selected, images will **not** be pushed to the private registry specified in the UI.
+- If online option is selected, updating Registry Settings page will **not** result in images being pushed.
+
+#### Fully airgapped mode
+
+In this case, both Admin Console and application are installed in airgapped mode:
+
+```
+kubectl kots install app-slug \
+  --kotsadm-registry my.private.registry/myapp \
+  --registry-username <username> \
+  --registry-password <password> \
+  --disable-image-push \
+  --airgap-bundle /path/to/app.airgap \
+  --license-file /path/to/license.yaml
+```
+
+When the above command is used, CLI will skip pushing images and proceed with the rest of the instalaltion.
+User will be redirect to the Config screen.
+Other command line arguments can be used for headless and/or unattended install experience without any additional changes.
+
+### Updates
+
+#### Online Admin Console
+
+Upgrading Admin Console installed in online mode does not change.
+
+#### Airgapped Admin Console
+
+Upgrading Admin Console installed in airgapped mode has to be done with the `--disable-image-push` flag specified on the command line:
```

Comment:
Right now registry credentials can be omitted in kURL installs only.  We'd need to move registry settings out of postgres since this command does not interact with the API.  I think this should be out of scope.  The flag could potentially be in a config map, which would not require the API.